libraries/ms-common/src/blink_event_generator.c

```diff
@@ -0,0 +1,68 @@
+#include "blink_event_generator.h"
+
+#define BLINK_EVENT_PRIORITY EVENT_PRIORITY_NORMAL
+
+static BlinkerState prv_opposite_state(BlinkerState state) {
+  return state == BLINKER_STATE_ON ? BLINKER_STATE_OFF : BLINKER_STATE_ON;
+}
+
+static uint16_t prv_state_to_value(BlinkerState state) {
+  return state == BLINKER_STATE_ON ? 1 : 0;
+}
+
+StatusCode blink_event_generator_init(BlinkEventGeneratorStorage *storage,
+                                      const BlinkEventGeneratorSettings *settings) {
+  if (settings->default_state >= NUM_BLINKER_STATES) {
+    return STATUS_CODE_INVALID_ARGS;
+  }
+
+  storage->interval_us = settings->interval_us;
+  storage->default_state = storage->current_state = settings->default_state;
+  storage->timer_id = SOFT_TIMER_INVALID_TIMER;
+  return STATUS_CODE_OK;
+}
+
+static bool prv_is_active(BlinkEventGeneratorStorage *storage) {
+  return storage->timer_id != SOFT_TIMER_INVALID_TIMER;
+}
+
+static void prv_raise_blink_event_callback(SoftTimerId timer_id, void *context) {
+  BlinkEventGeneratorStorage *storage = context;
+  BlinkerState new_state = prv_opposite_state(storage->current_state);
+  event_raise_priority(BLINK_EVENT_PRIORITY, storage->event_id, prv_state_to_value(new_state));
+  storage->current_state = new_state;
+
+  soft_timer_start(storage->interval_us, &prv_raise_blink_event_callback, storage,
+                   &storage->timer_id);
+}
+
+StatusCode blink_event_generator_start(BlinkEventGeneratorStorage *storage, EventId event_id) {
+  if (prv_is_active(storage)) {
+    if (event_id == storage->event_id) {
+      // restarting with the same event: do nothing to avoid a strange delay
```

Comment:
Actually, I think restarting the timer here would be valuable. By that I mean just simply cancelling the old timer and starting a new timer. 

The reason this'll be good is because when we want to sync the rear lights and the front lights, the rear power dist board will send a `sync` message to the front board, and the front board can just simply call `start` and that will do the resetting. That way our lights will also be in sync after longer periods of time.